Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -730,13 +730,21 @@ extension NavigationViewController: NavigationServiceDelegate {
         
         arrivalController?.showEndOfRouteIfNeeded(self,
                                                   advancesToNextLeg: advancesToNextLeg,
-                                                  completion: nil,
+                                                  completion: { [ weak self] _ in
+                                                    self?.frameDestinationArrival(for: service.router.location)
+                                                  },
                                                   onDismiss: { [weak self] in
                                                     self?.navigationService.endNavigation(feedback: $0)
                                                     self?.handleCancelAction()
                                                   })
         return advancesToNextLeg
     }
+    
+    func frameDestinationArrival(for location: CLLocation?) {
+        if waypointStyle == .annotation { return }
+        guard let location = location else { return }
```

Comment:
```suggestion
        guard waypointStyle != .annotation, let location = location else { return }
```

Reply:
I guess guard should be used for early return.

Reply:
Thanks for the suggestion, I'll improve this part.